### PR TITLE
Add `completion.item_padding`

### PR DESCRIPTION
**Pull request by kyza0dev**

### CODE REVIEW

File: qutebrowser/completion/completionwidget.py [hunk 1]
```diff
@@ -78,6 +78,13 @@ class CompletionView(QTreeView):
                 {{ conf.colors.completion.category.border.bottom }};
         }
 
+        QTreeView:item {
+            padding-top: {{ completion.item_padding['top'] }}px;
+            padding-left: {{ completion.item_padding['left'] }}px;
+            padding-right: {{ completion.item_padding['right'] }}px;
+            padding-bottom: {{ completion.item_padding['bottom'] }}px;
```
Comment (The-Compiler) [inline qutebrowser/completion/completionwidget.py:85]:
```suggestion
            padding-top: {{ conf.completion.item_padding['top'] }}px;
            padding-left: {{ conf.completion.item_padding['left'] }}px;
            padding-right: {{ conf.completion.item_padding['right'] }}px;
            padding-bottom: {{ conf.completion.item_padding['bottom'] }}px;
```

**Review by The-Compiler — CHANGES_REQUESTED** [review]
This just crashes qutebrowser outright. Please make sure you actually try your code and can confirm it works.

File: qutebrowser/config/configdata.yml [hunk 2]
```diff
@@ -1267,6 +1267,15 @@ completion.scrollbar.padding:
     minval: 0
   desc: Padding (in pixels) of the scrollbar handle in the completion window.
 
+completion.item_padding:
+  type: Padding
```
Comment (segf00lt) [inline qutebrowser/config/configdata.yml:1271]: `type` should be:
```
type:
  name: Dict
  keytype: String
  valtype:
    name: Int
    minval: 0
```

**Comment by segf00lt** [comment]
For controling the padding values it might be better to use separate options instead of a Dict.
```
completion.item_padding.top:
  default: 1
  type:
    name: Int
    minval: 0
  desc: Top padding (in pixels) of items in the completion window.

completion.item_padding.bottom:
  default: 1
  type:
    name: Int
    minval: 0
  desc: Bottom padding (in pixels) of items in the completion window.

completion.item_padding.left:
  default: 0
  type:
    name: Int
    minval: 0
  desc: Left padding (in pixels) of items in the completion window.

completion.item_padding.right:
  default: 0
  type:
    name: Int
    minval: 0
  desc: Right padding (in pixels) of items in the completion window.
```

And the changes for the stylesheet in completionwidget.py:
```
        QTreeView:item {
            padding-top: {{ conf.completion.item_padding.top }}px;
            padding-bottom: {{ conf.completion.item_padding.bottom }}px;
            padding-left: {{ conf.completion.item_padding.left }}px;
            padding-right: {{ conf.completion.item_padding.right }}px;
        }
```

**Comment by The-Compiler** [comment]
Why? We already have a `Padding` type we use elsewhere, so it seems to make sense to use it here too, no?

**Comment by segf00lt** [comment]
Yeah that makes more sense. I was thinking it would be a setting the user would occasionally change at runtime, but just as a setting in the config file it does make more sense to use `Padding`.

**Comment by vereis** [comment]
This looks like it hasn't had any updates in awhile, @The-Compiler would you be open to another shot at implementing this feature?

**Comment by The-Compiler** [comment]
@vereis Sorry for the delay, I was taking a bit of a break from qutebrowser stuff. I *think* this one should actually be okay from a quick look, unless you see something that would be missing here.

Usually with older PRs it's really appreciated if someone helps picking things up and getting them updated, but here I think there is nothing left to do, except on my end of things. I'm currently focusing on Qt 6 stuff, but then want to go back and hopefully merge a big batch of PRs again after that.

**Review by mschilli87 — APPROVED** [review]
`configdata.yml` changes LGTM as well.

**Comment by vereis** [comment]
No worries at all! I really appreciate it; I've only started using qutebrowser recently but I'm really loving it so far!

Really appreciate you following up again :heart: Looking forward to everything else you have in store!

File: qutebrowser/config/configdata.yml [hunk 3]
```diff
@@ -1267,6 +1267,15 @@ completion.scrollbar.padding:
...
+  default:
+    top: 1
+    bottom: 1
+    left: 0
+    right: 0
```
Comment (arza-zara) [inline qutebrowser/config/configdata.yml:1276]: I somewhat prefer the old default of 0 for all

**Review by arza-zara — APPROVED** [review]

**Comment by mschilli87** [comment]
@The-Compiler: Just making sure this doesn't slip through your attention due to @kyza0dev removing their fork.

**Pull request closed without merging**